### PR TITLE
Signup view

### PR DESCRIPTION
**Pull request by JoseBiescas**
modified signin api route from login to signin

User sign in and sign up functionality added
Password hashing added
Email confirmation added

Co-authored-by: chris810 <chris810@users.noreply.github.com>

### CODE REVIEW

File: backend/api/handler/account_handler.py [hunk 1]
```diff
@@ -100,3 +105,65 @@ def update_account(uid,json):
...
+            account_dao = accountDAO.get_account_email(json['email'])
+            if account_dao:
+                return jsonify(reason="Account with given email already exists."), HttpStatus.BAD_REQUEST.value
+            username_exists = accountDAO.get_account_username(json['username'])
+            if username_exists:
+                return jsonify(reason="Username already taken"), HttpStatus.BAD_REQUEST.value
+            create_account = accountDAO.create_account(json)
+            token = AccountHandler.generate_confirmation_token(json.get('email'))
+            send_email(json['email'], "Email Confirmation Code", token)
+            # confirm_url = url_for('user.confirm_email', token=token, _external=True)
```
Comment (chris810) [inline backend/api/handler/account_handler.py:134]: nitpick remove these comments.

File: signy/src/features/signin/signinSlice.ts [hunk 2]
```diff
@@ -29,14 +46,25 @@ export const signinSlice = createSlice({
...
-        builder.addCase(signinAsync.fulfilled, (state) => {
-            state.isAuth = !state.isAuth;
+        builder.addCase(signinAsync.rejected, (state, action) => {
+            console.log(action.error);
+        })
+        builder.addCase(signinAsync.fulfilled, (state, action) => {
+            state.user = action.payload;
+            state.isAuth = true;
+        })
+        builder.addCase(signupAsync.rejected, (state, action) => {
```
Comment (chris810) [inline signy/src/features/signin/signinSlice.ts:56]: Question, .rejected y .fulfilled son parte de redux?

**Review by chris810 — COMMENTED** [review]
LGTM! we pair programmed parts of it, so skimmed through that. I left a comment asking about a functionality of Redux.

File: signy/src/features/signin/signinSlice.ts (hunk 2 above)
Comment (JoseBiescas) [reply signy/src/features/signin/signinSlice.ts:56]: Yes

https://redux-toolkit.js.org/api/createAsyncThunk#type

File: signy/src/features/signup/SignUp.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    )
+  } else {
+      content = (
+        <div className={styles.row}>
+        <div className={styles.signupBox}>
+          <div className={styles.heading}>
+            <h1 className={styles.h1}>Sign Up</h1>
+          </div>
+          <form className={styles.form}>
+            <label className={styles.label}>Firstname</label>
```
Comment (chris810) [inline signy/src/features/signup/SignUp.tsx:91]: missing space.

**Pull request merged**